flask/boxwise_flask/models/box.py

```diff
@@ -110,38 +114,38 @@ def __unicode__(self):
         return self.box_label_identifier
 
     @staticmethod
-    def create_box(box_creation_input):
+    def get_box(box_id):
+        return Box.get(Box.box_label_identifier == box_id)
 
-        today = datetime.now()
-        barcode = box_creation_input.get("qr_barcode", None)
-        qr_id_from_table = QRCode.get_id_from_code(barcode)
-        box_uuid = uuid.uuid4()
-        box_short_uuid = str(box_uuid)[
-            :11
-        ]  # the table is truncating a full uuid to 11 chars, so do it preemptively
 
-        new_box = Box.create(
-            # surprisingly not primary key, unique non-sequential identifier for a box
-            box_label_identifier=box_short_uuid,
-            product_id=box_creation_input.get(
-                "product_id", None
-            ),  # will become a fancy dropdown on the FE
-            size_id=box_creation_input.get(
-                "size_id", None
-            ),  # will be tied to the product_id lookup somehow
-            items=box_creation_input.get("items", None),
-            location_id=box_creation_input.get(
-                "location_id", None
-            ),  # based on the user's allowed bases
-            comment=box_creation_input.get("comment", None),
-            qr_id=qr_id_from_table,
-            created=today,
-            # this is consistently NULL in the table, do we want to change that?
-            created_by=box_creation_input.get("created_by", None),
-            box_state_id=1,  # always 1 for create?
-        )
-        return new_box
+def create_box(box_creation_input):
+    """Insert information for a new Box in the database. Use current datetime
+    and box state "InStock" by default. Generate a UUID to identify the box.
+    """
+    now = datetime.now()
+    qr_code = box_creation_input.pop("qr_code", None)
+    qr_id = QRCode.get_id_from_code(qr_code)
 
-    @staticmethod
-    def get_box(box_id):
-        return Box.get(Box.box_label_identifier == box_id)
+    new_box = Box.create(
+        box_label_identifier=str(uuid.uuid4())[: Box.box_label_identifier.max_length],
+        qr_id=qr_id,
+        created_on=now,
```

Comment:
@aerinsol Do we consider creation a modification? For consistency we should then set the `last_modified_*` at creation, too, even if the values are identical to the `created_*` fields.

Reply:
Yes, set it identical to the create field. @pylipp